src/main/java/org/opensearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -48,13 +49,7 @@ private static void requireEqualDimension(final float[] queryVector, final float
      * @return L2 score
      */
     public static float l2Squared(float[] queryVector, float[] inputVector) {
-        requireEqualDimension(queryVector, inputVector);
-        float squaredDistance = 0;
-        for (int i = 0; i < inputVector.length; i++) {
-            float diff = queryVector[i] - inputVector[i];
-            squaredDistance += diff * diff;
-        }
-        return squaredDistance;
+        return VectorUtil.squareDistance(queryVector, inputVector);
```

Comment:
should we requireEqualDim here too?

Reply:
@jmazanec15 Lucene already have the support for it